openslides_backend/action/meeting/delete_all_speakers_of_all_lists.py

```diff
@@ -0,0 +1,36 @@
+from ...models.models import Speaker
+from ...shared.exceptions import ActionException
+from ...shared.patterns import Collection, FullQualifiedId
+from ..default_schema import DefaultSchema
+from ..generics import ActionPayload, DeleteAction
+from ..register import register_action
+
+
+@register_action("meeting.delete_all_speakers_of_all_lists")
+class DeleteAllSpeakersOfAllListsAction(DeleteAction):
+    """
+    Action to delete all speakers of all lists of a meeting.
+    """
+
+    model = Speaker()  # we want to delete Speakers
+    schema = DefaultSchema(Speaker()).get_delete_schema()
+
+    def update_instances(self, payload: ActionPayload) -> ActionPayload:
+        new_payload = []
+        for instance in payload:
+            meeting = self.database.get(
+                FullQualifiedId(Collection("meeting"), instance["id"]),
+                ["list_of_speakers_ids"],
+            )
+            if not meeting.get("list_of_speakers_ids"):
+                continue
+            for los in meeting["list_of_speakers_ids"]:
+                list_of_speakers = self.database.get(
+                    FullQualifiedId(Collection("list_of_speakers"), los),
+                    ["speaker_ids"],
+                )
+                for speaker in list_of_speakers.get("speaker_ids", []):
+                    new_payload.append({"id": speaker})
+        if not new_payload:
+            raise ActionException("No speakers to delete.")
```

Comment:
See #247

Reply:
The underlying code cannot handle a empty payload case.

def merge_write_request_elements(
       [...]
        if user_id is None:
>           raise ValueError("At least one of the given user ids must not be None.")
E           ValueError: At least one of the given user ids must not be None.

openslides_backend/action/base.py:384: ValueError

Reply:
Here we either have to insist in sending HTTP 400 or we have to add a new exception that is parsed to HTTP 200 or even better HTTP 204

Reply:
Add an exception which returns a  HTTP 200.

Reply:
...or we could update the merge method to be able to handle empty write events? Throwing an exception in a success case seems really hacky. The only other thing to check is then at the end, if eyn events should be written to the datastore at all, which should be easy.

Reply:
Sounds reasonable.